godot-core/src/engine/script_instance.rs

```diff
@@ -505,23 +511,32 @@ mod script_instance_info {
     ) -> *const sys::GDExtensionPropertyInfo {
         let ctx = || format!("error when calling {}::get_property_list", type_name::<T>());
 
-        let property_list = handle_panic(ctx, || {
+        let (property_list, property_sys_list) = handle_panic(ctx, || {
             let instance = instance_data_as_script_instance::<T>(p_instance);
 
-            let property_list: Vec<_> = borrow_instance(instance)
-                .get_property_list()
+            let property_list = borrow_instance(instance).get_property_list();
+
+            let property_sys_list: Box<[_]> = property_list
                 .iter()
                 .map(|prop| prop.property_sys())
                 .collect();
 
-            property_list
+            (property_list, property_sys_list)
         })
         .unwrap_or_default();
 
         // SAFETY: list_length has to be a valid pointer to a u32.
         let list_length = unsafe { &mut *r_count };
+        let return_pointer = transfer_ptr_list_to_godot(property_sys_list, list_length);
+
+        let instance = instance_data_as_script_instance::<T>(p_instance);
+        instance
+            .property_list
+            .lock()
+            .expect("Mutex should not be poisoned")
+            .insert(return_pointer, property_list);
```

Comment:
What about using an `Arc` instead of a Box? It would allow the implementor to reuse an existing slice if they wish to. I have never tried this and don't know how `Arc`s and raw pointers interact, but I assume you might know.

Reply:
I'm not entirely sure, but i dont think they have the same issue as box. `Box` is rather special in rust so has a bunch of extra rules that `Arc` shouldn't have. I think it'd be easier to use `Vec` though. 

If you have a particular reason it'd be better to use `Arc` than `Vec` then feel free to. But barring a use-case i think `Vec` is preferable since it's gonna be more familiar to more people.

Reply:
Miri seems happy with `Arc`:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=21b9778a61c05f69b90f6de2a841200f

Reply:
Use case would be to be able to reuse the same slice multiple times instead of returning a new one every call.

Reply:
Using an Arc unfortunately causes a Clippy warning because `PropertyInfo` / `MethodInfo` is neither `Send` nor `Sync`. I think for the use case here that is not actually an issue, but this warning is raised at every `Arc::new([...])` so also in user code. I'm sticking with `Vec<PropertyInfo>`.

Reply:
What about `Rc`?

I also think there's value in familiarity, and `Vec` is easier to use. Question is whether these functions are invoked so often that it could cause a problem...

Reply:
@Bromeon the `Rc` could be cloned in `get_property_list_func` and will be dropped in `free_property_list_func` which can be called by the engine on multiple threads, so the reference count decrement would not be thread safe.